### ISSUE TITLE
aws-vault session ends and credentials aren't refreshed

### ISSUE THREAD

**Issue by daveyu** [cloudposse/geodesic#350]
## what
When using `assume-role`, the `aws-vault` session times out (after 1hr by default), and credentials are not refreshed. User is prompted to `assume-role` again but `aws-vault` fails to start the credential server.
```
[2]+  Stopped                 aws-vault exec ${AWS_VAULT_ARGS[@]} $role -- bash -l
-> Run 'assume-role' to login to AWS
 ⧉  testing
 ✗   (none) ~ ⨠  assume-role
Enter passphrase to unlock /conf/.awsvault/keys/:
aws-vault: error: Failed to start credential server: listen tcp 127.0.0.1:9099: bind: address already in use
```

## why
Expected behavior: user only needs to `assume-role` once and `aws-vault` silently refreshes credentials for the aws session.

**Comment by daveyu** [comment]
More context and initial triaging from slack: https://sweetops.slack.com/archives/CB84E9V54/p1546897392022500

**Comment by daveyu** [comment]
Some debug info for after the session dies:
```
[2]+  Stopped                 aws-vault exec ${AWS_VAULT_ARGS[@]} $role -- bash -l
```
```
-> Run 'assume-role' to login to AWS
 ⧉  testing
 ✗   (none) ~ ⨠  ps aux
PID   USER     TIME  COMMAND
    1 root      0:00 /bin/bash -l
   16 root      0:00 aws-vault server
   54 root      0:00 ssh-agent
   58 root      0:00 {syslog-ng} supervising syslog-ng
   59 root      0:00 syslog-ng -f /etc/syslog-ng/syslog-ng.conf
  109 root      0:00 aws-vault exec --assume-role-ttl=1h --server lootcrate-testing-admin -- bash -l
  121 root      0:00 bash -l
  501 root      0:00 ps aux
```
```
-> Run 'assume-role' to login to AWS
 ⧉  testing
 ✗   (none) ~ ⨠  lsof -iTCP:9099
1	/bin/bash	/dev/pts/0
1	/bin/bash	/dev/pts/0
1	/bin/bash	/dev/pts/0
1	/bin/bash	/dev/pts/0
16	/usr/bin/aws-vault	/dev/null
16	/usr/bin/aws-vault	/dev/pts/0
16	/usr/bin/aws-vault	/dev/pts/0
16	/usr/bin/aws-vault	socket:[750666]
16	/usr/bin/aws-vault	anon_inode:[eventpoll]
16	/usr/bin/aws-vault	socket:[830398]
16	/usr/bin/aws-vault	socket:[830400]
16	/usr/bin/aws-vault	socket:[833622]
16	/usr/bin/aws-vault	socket:[833624]
54	/usr/bin/ssh-agent	/dev/null
54	/usr/bin/ssh-agent	/dev/null
54	/usr/bin/ssh-agent	/dev/null
54	/usr/bin/ssh-agent	socket:[750676]
58	/usr/sbin/syslog-ng	/dev/null
58	/usr/sbin/syslog-ng	/dev/null
58	/usr/sbin/syslog-ng	/dev/null
59	/usr/sbin/syslog-ng	/dev/null
59	/usr/sbin/syslog-ng	/dev/null
59	/usr/sbin/syslog-ng	/dev/null
59	/usr/sbin/syslog-ng	anon_inode:[eventpoll]
59	/usr/sbin/syslog-ng	pipe:[746383]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	socket:[747515]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	anon_inode:[eventfd]
59	/usr/sbin/syslog-ng	/run/syslog-ng.persist
59	/usr/sbin/syslog-ng	socket:[752641]
109	/usr/bin/aws-vault	/dev/pts/0
109	/usr/bin/aws-vault	/dev/pts/0
109	/usr/bin/aws-vault	/dev/pts/0
109	/usr/bin/aws-vault	socket:[832959]
109	/usr/bin/aws-vault	anon_inode:[eventpoll]
109	/usr/bin/aws-vault	socket:[746455]
121	/bin/bash	/dev/pts/0
121	/bin/bash	/dev/pts/0
121	/bin/bash	/dev/pts/0
121	/bin/bash	/dev/pts/0
464	/usr/bin/chamber	/dev/pts/0
464	/usr/bin/chamber	/dev/pts/0
464	/usr/bin/chamber	/dev/pts/0
464	/usr/bin/chamber	socket:[830414]
464	/usr/bin/chamber	anon_inode:[eventpoll]
```
```
-> Run 'assume-role' to login to AWS
 ⧉  testing
 ✗   (none) ~ ⨠  ps aux
PID   USER     TIME  COMMAND
    1 root      0:00 /bin/bash -l
   16 root      0:00 aws-vault server
   54 root      0:00 ssh-agent
   58 root      0:00 {syslog-ng} supervising syslog-ng
   59 root      0:00 syslog-ng -f /etc/syslog-ng/syslog-ng.conf
  109 root      0:00 aws-vault exec --assume-role-ttl=1h --server lootcrate-testing-admin -- bash -l
  121 root      0:00 bash -l
  564 root      0:00 ps aux
```
```
 ✗   (none) ~ ⨠  Error: Failed to list store contents: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```

**Comment by Nuru** [comment]
Closing this as it was mostly addressed in #383, extending the life of the session to 12 hours. The issue of how to handle the end of an `aws-vault` session at the end of 12 hours is very difficult due to the way `bash` job control works, and is no longer that important since it only comes up once every 12 hours.

**Comment by Nuru** [comment]
The remaining issue of having `aws-vault` continue to work at the end of the 12 hour session is up to them. See https://github.com/99designs/aws-vault/issues/362
